seatunnel-core/seatunnel-flink-starter/src/main/java/org/apache/seatunnel/core/starter/flink/execution/FlinkTaskExecution.java

```diff
@@ -46,7 +47,8 @@ public class FlinkTaskExecution {
 
     public FlinkTaskExecution(Config config) {
         this.config = config;
-        this.flinkEnvironment = createFlinkEnvironment();
+        this.flinkEnvironment = (FlinkEnvironment) new EnvironmentFactory<>(config, EngineType.FLINK).getEnvironment();
```

Comment:
Please rollback this file.

Reply:
I change the EnvironmentFactory and FlinkEnvironment to support change `job.mode` and create `StreamExecutionEnvironment`. So this method will work, this way have any problem?

Reply:
OK, I see your change in `EnvironmentFactory`, the old `EnvironmentFactory` has some problem in new API.